core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -348,10 +348,7 @@ sealed abstract class P2WPKHSigner extends BitcoinSigner[P2WPKHV0SpendingInfo] {
                                                unsignedTxWitness)
 
           val witSPK = output.scriptPubKey match {
-            case p2wpkh: P2WPKHWitnessSPKV0 =>
-              if (p2wpkh != P2WPKHWitnessSPKV0(pubKey)) {
-                Future.fromTry(TxBuilderError.WrongPublicKey)
-              } else Future.successful(p2wpkh)
+            case p2wpkh: P2WPKHWitnessSPKV0 => Future.successful(p2wpkh)
```

Comment:
Is this condition being checked somewhere else?

Reply:
It is, that is now `P2WPKHSigner`'s responsibility